lib/stripe/core_resources/payout.ex

```diff
@@ -48,4 +49,116 @@ defmodule Stripe.Payout do
+               amount: pos_integer,
+               currency: String.t(),
+               description: String.t(),
+               destination: %{
+                 :account => Stripe.id() | Stripe.Account.t(),
+                 optional(:amount) => non_neg_integer
+               },
+               metadata: map,
+               source_type: String.t(),
+               statement_descriptor: String.t()
+             }
+  def create(params, opts \\ []) do
+    new_request(opts)
+    |> put_endpoint(@plural_endpoint)
+    |> put_params(params)
+    |> put_method(:post)
+    |> make_request()
+  end
+
+  @doc """
+  Retrieve a payout.
+
+  See the [Stripe docs](https://stripe.com/docs/api#retrieve_payout).
+  """
+  @spec retrieve(Stripe.id() | t, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
+  def retrieve(id, opts \\ []) do
+    new_request(opts)
+    |> put_endpoint(@plural_endpoint <> "/#{get_id!(id)}")
+    |> put_method(:get)
+    |> make_request()
+  end
+
+  @doc """
+  Update a payout.
+
+  Updates the specified payout by setting the values of the parameters passed. Any parameters
+  not provided will be left unchanged.
+
+  This request accepts only the `:payout` or `:metadata`.
+
+  The payout to be updated may either be passed in as a struct or an ID.
+
+  See the [Stripe docs](https://stripe.com/docs/api#update_payout).
+  """
+  @spec update(Stripe.id() | t, params, Stripe.options()) :: {:ok, t} | {:error, Stripe.Error.t()}
+        when params: %{
+               metadata: %{
+                 optional(String.t()) => String.t(),
+                 optional(atom) => String.t()
+               }
+             }
+  def update(id, params, opts \\ []) do
+    new_request(opts)
+    |> put_endpoint(@plural_endpoint <> "/#{get_id!(id)}")
+    |> put_method(:post)
+    |> put_params(params)
+    |> make_request()
+  end
+
+  @doc """
+  List all payouts.
+
+  Returns a list of payouts. The payouts are returned in sorted order,
+  with the most recent payouts appearing first.
+
+  See the [Stripe docs](https://stripe.com/docs/api#list_payouts).
+  """
+  @spec list(params, Stripe.options()) :: {:ok, Stripe.List.of(t)} | {:error, Stripe.Error.t()}
+        when params: %{
+               arrival_date: Stripe.date_query(),
+               created: Stripe.date_query(),
+               destination: String.t(),
+               ending_before: t | Stripe.id(),
+               limit: 1..100,
+               starting_after: t | Stripe.id(),
+               status: String.t()
+             }
+  def list(params \\ %{}, opts \\ []) do
+    new_request(opts)
+    |> put_endpoint(@plural_endpoint)
+    |> put_method(:get)
+    |> put_params(params)
+    |> cast_to_id([:ending_before, :starting_after])
+    |> make_request()
+  end
+
+  @doc """
+  Cancel a payout.
+
+  See the [Stripe docs](https://stripe.com/docs/api#cancel_payout).
+  """
+  @spec cancel(Stripe.id() | t) :: {:ok, t} | {:error, Stripe.Error.t()}
+  def cancel(id) do
```

Comment:
This should take `Stripe.options()` as the second param like the other functions.